connectors/sources/sharepoint.py

```diff
@@ -283,27 +303,40 @@ async def _api_call(self, url_name, url="", **url_kwargs):
                     else:
                         yield await result.json()
                     break
-            except Exception as exception:
-                if isinstance(
-                    exception,
-                    ClientResponseError,
-                ) and "token has expired" in exception.headers.get(  # pyright: ignore
+            except ClientResponseError as exception:
+                retry, retry_seconds = self._get_retry_after(
+                    retry=retry, exception=exception
+                )
+                if getattr(exception, "status", None) == 429:
+                    logger.warning("Rate limit exceeded.")
```

Comment:
Please add additional information about rate limit into warning - what data came from the Sharepoint and how it affected the run?